vault/external_tests/api/kv_helpers_test.go

```diff
@@ -295,5 +296,137 @@ func TestKVHelpers(t *testing.T) {
+		}
+
+		// get its metadata to make sure it was created successfully
+		md, err := client.KVv2(mountPath).GetMetadata(context.Background(), noDataSecretPath)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if md.CreatedTime.IsZero() {
+			t.Fatalf("secret metadata was not populated as expected: %v", err)
+		}
+		if len(md.Versions) > 0 {
+			t.Fatalf("no secret versions should have been created since only metadata was populated")
+		}
+
+		// replace all modifiable metadata fields
+		casRequired := true
+		deleteVersionAfter = 6 * time.Hour
+		maxVersions = 6
+		customMetadata = map[string]interface{}{"ape": "orangutan", "cat": "tabby"}
+		err = client.KVv2(mountPath).PutMetadata(context.Background(), noDataSecretPath, api.KVMetadataInput{
+			CASRequired:        &casRequired,
+			DeleteVersionAfter: &deleteVersionAfter,
+			MaxVersions:        &maxVersions,
+			CustomMetadata:     customMetadata,
+		})
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// check that metadata was replaced
+		md2, err := client.KVv2(mountPath).GetMetadata(context.Background(), noDataSecretPath)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if md.CASRequired == md2.CASRequired || md.MaxVersions == md2.MaxVersions || md.DeleteVersionAfter == md2.DeleteVersionAfter || reflect.DeepEqual(md.CustomMetadata, md2.CustomMetadata) {
+			t.Fatalf("metadata fields should have been updated by PutMetadata")
+		}
+
+		// now let's try a patch
+		maxVersions = 7
+		err = client.KVv2(mountPath).PatchMetadata(context.Background(), noDataSecretPath, api.KVMetadataInput{
+			MaxVersions:    &maxVersions,
+			CustomMetadata: map[string]interface{}{"ape": nil, "rat": "brown"},
+		})
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		// check that the metadata was only partially replaced
+		md3, err := client.KVv2(mountPath).GetMetadata(context.Background(), noDataSecretPath)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if md2.CASRequired != md3.CASRequired || md2.DeleteVersionAfter != md3.DeleteVersionAfter {
+			t.Fatalf("expected fields to remain unchanged but they were updated")
+		}
+		if md3.MaxVersions == 0 {
+			t.Fatalf("field was reset to its zero value when it should not have been")
+		}
+		if md2.MaxVersions == md3.MaxVersions {
+			t.Fatalf("expected field to be updated but it remained unchanged")
+		}
+
+		// let's check the custom metadata was updated correctly
+		if r, ok := md3.CustomMetadata["rat"]; ok {
+			if r != "brown" {
+				t.Fatalf("expected value to be \"brown\"")
+			}
+		} else {
+			t.Fatalf("expected there to be a new \"rat\" key")
+		}
+
+		if _, ok := md3.CustomMetadata["ape"]; ok {
+			t.Fatalf("expected \"ape\" key to be removed")
+		}
+
+		if _, ok := md3.CustomMetadata["cat"]; !ok {
+			t.Fatalf("did not expect \"cat\" key to be removed")
+		}
+
+		// now let's do another patch to test the "explicit zero value" use case
+		explicitFalse := false
+		explicitZero := 0
+		explicitTimeZero, err := time.ParseDuration("0s")
+		if err != nil {
+			t.Fatal(err)
+		}
```

Comment:
```suggestion
		var (
			explicitFalse    bool
			explicitZero     int
			explicitTimeZero time.Duration
		)
```
I think this would give the same effect of default-initializing the values.